README.md

```diff
@@ -1545,6 +1551,79 @@ all:
   monitoring. Defaults to empty list if not set. Always refer to the devices
   using the long, stable device name (`/dev/disk/by-id/`).
 
+## Export current cluster configuration
+
+The role provides `ha_cluster_info` module which exports current cluster
+configuration in a dictionary matching the structure of this role variables. If
+the role is run with these variables, it recreates the same cluster.
+
+Note that the dictionary of variables may not be complete and manual
+modification of it is expected. Most notably, you need to set
+[`ha_cluster_hacluster_password`](#ha_cluster_hacluster_password).
+
+Note that depending on pcs version installed on managed nodes, certain variables
+may not be present in the export.
+
+* Following variables are present in the export:
+  * [`ha_cluster_cluster_present`](#ha_cluster_cluster_present)
+  * [`ha_cluster_start_on_boot`](#ha_cluster_start_on_boot)
+  * [`ha_cluster_cluster_name`](#ha_cluster_cluster_name)
+  * [`ha_cluster_transport`](#ha_cluster_transport)
+  * [`ha_cluster_totem`](#ha_cluster_totem)
+  * [`ha_cluster_quorum`](#ha_cluster_quorum)
+  * [`ha_cluster_node_options`](#ha_cluster_node_options) - currently only
+    `node_name`, `corosync_addresses` and `pcs_address` are present
+
+* Following variables are never present in the export (consult the role
+  documentation for impact of the variables missing when running the role):
+  * [`ha_cluster_hacluster_password`](#ha_cluster_hacluster_password) - This is
+    a mandatory variable for the role but it cannot be extracted from existing
+    clusters.
+  * [`ha_cluster_corosync_key_src`](#ha_cluster_corosync_key_src),
+    [`ha_cluster_pacemaker_key_src`](#ha_cluster_pacemaker_key_src) and
+    [`ha_cluster_fence_virt_key_src`](#ha_cluster_fence_virt_key_src) - These
+    are supposed to contain paths to files with the keys. Since the keys
+    themselves are not exported, these variables are not present in the export
+    either. Corosync and pacemaker keys are supposed to be unique for each
+    cluster.
+  * [`ha_cluster_regenerate_keys`](#ha_cluster_regenerate_keys) - It is your
+    responsibility to decide if you want to use existing keys or generate new
+    ones.
+
+To export current cluster configuration and store it in
+`ha_cluster_info_result` variable, write a task like this:
+
+```yaml
+- name: Get current cluster configuration
+  linux-system-roles.ha_cluster.ha_cluster_info:
+  register: ha_cluster_info_result
```

Comment:
System roles, by convention, do not support users using modules directly.  In every other role that does something like this, users use the role with either no arguments like https://github.com/linux-system-roles/firewall?tab=readme-ov-file#gathering-firewall-ansible-facts:

```yaml
- name: Get current cluster configuration
  include_role:
    name: linux-system-roles.ha_cluster
```

or with some special variable

```yaml
- name: Get current cluster configuration
  include_role:
    name: linux-system-roles.ha_cluster
  vars:
    ha_cluster_get_info: true
```

I think the ha_cluster role will have to do something like the latter, since there are numerous public api variables, as opposed to the firewall role which just has the one main `firewall` variable.  The latter also makes it possible for the role to

* set the state of the cluster *and* return the cluster configuration
* return a subset of the information

Reply:
The role would then set a *global* variable e.g. `ha_cluster_info` that users would use.  This return variable will be declared in the README.md in the section `Variables Exported by the Role` e.g. https://github.com/linux-system-roles/kernel_settings?tab=readme-ov-file#variables-exported-by-the-role

Reply:
ping

Reply:
bootloader and snapshot roles export info with `<rolename>_facts` variable. Let's be consistent with this naming.

Reply:
Sorry, I've been busy with other projects.

I wasn't sure what would be the correct way to expose the export functionality. So I'm glad you pointed me in the right direction. I'm going to implement this change, hopefully in a couple of weeks, once I finish tasks that require my immediate attention.

Reply:
I pushed new commits which implement this requested change.